editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+        resultOfApplyEdits.addDoneListener(() -> {
+          try {
+            // check if the server edit was successful
+            List<FeatureTableEditResult> edits = resultOfApplyEdits.get();
+            if (edits == null || edits.size() == 0) {
+              new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+            } else {
+              // if the edits were successful, switch to the default version
+              switchVersion(defaultVersion);
+            }
+          } catch (InterruptedException | ExecutionException e) {
+            new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+          }
+        });
+      } else {
+        // if there are no local edits, switch to the default version
+        switchVersion(defaultVersion);
+      }
+    } else if (serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+      // if the current version is the default version, switch to the user created version
+      switchVersion(userCreatedVersion);
+    }
+  }
+
+  /**
+   * Switch the active branch version
+   *
+   * @param versionName name of the version to switch to
+   */
+  private void switchVersion(String versionName) {
+    ListenableFuture<Void> switchVersionResult = serviceGeodatabase.switchVersionAsync(versionName);
+    switchVersionResult.addDoneListener(() -> {
+      // check if the active version has switched successfully and update the UI
+      if (serviceGeodatabase.getVersionName().equals(versionName)) {
+        currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+        editFeatureVBox.setDisable(true);
+      } else {
+        new Alert(Alert.AlertType.ERROR, "Error switching version.").show();
+      }
+    });
+  }
+
+  /**
+   * Select a feature if one exists where the user clicked
+   *
+   * @param point location where the user clicked
+   */
+  private void selectFeature(Point2D point){
+    ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, point, 1, false);
+    results.addDoneListener(() -> {
+      try {
+        IdentifyLayerResult layer = results.get();
+        List<GeoElement> identified = layer.getElements();
+        if (!identified.isEmpty()) {
+          GeoElement element = identified.get(0);
+          if (element instanceof ArcGISFeature) {
+            // get the selected feature
+            selectedFeature = (ArcGISFeature) element;
+            featureLayer.selectFeature(selectedFeature);
+            selectedFeature.loadAsync();
+            selectedFeature.addDoneLoadingListener(() -> {
+              if (selectedFeature.getLoadStatus() == LoadStatus.LOADED) {
+                // when a feature has been selected set the damage combo box to the feature's attribute value
+                damageTypeComboBox.getSelectionModel().select((String) selectedFeature.getAttributes().get("typdamage"));
```

Comment:
This is quite a long statement, perhaps it'd be easier to recognise what's happening if we split it a bit:
```suggestion
				String attributeValue = (String) selectedFeature.getAttributes().get("typdamage");
                damageTypeComboBox.getSelectionModel().select(attributeValue);
```

Overall this seems like quite a 'risky' way of correlating between the `attribute` and the item in the ComboBox. We'd have to know for 100% certainty that the result of the attribute exists within the combobox, otherwise the app would crash with quite a nasty error. 
Might be worth thinking about adding a method that checks `attributeValue` against the options available in the ComboBox, and returns an error if it doesn't exist? We know this wouldn't happen in our sample, but should a user borrow this code for their app, we might want to make sure there's some validation (which would be 'best practice'). If it complicates the sample too much, then it should be ok to just leave it.

Reply:
Have added the additional check here and also as a result pulled out the `getAttributes` since it's now used twice. So see what you think....